### PR TITLE
Add support for custom data api location for the deploy command

### PR DESCRIPTION
**Pull request by thomasgauvin**
Data API Location can be specified via cli arg, config file parameter, workflow file parameter

### CODE REVIEW

File: src/cli/commands/deploy/deploy.ts [hunk 1]
```diff
@@ -34,6 +34,11 @@ export async function deploy(options: SWACLIConfig) {
   // make sure appLocation is set
   appLocation = path.resolve(appLocation || process.cwd());
 
+  // make sure dataApiLocation is set
```
Comment (vivekjilla) [inline src/cli/commands/deploy/deploy.ts:37]: I think we also use some default folder name for data-api-location. Do we also need to pass that folder name in default scenarios.

File: src/cli/commands/deploy/deploy.ts [hunk 2]
```diff
@@ -34,6 +34,11 @@ export async function deploy(options: SWACLIConfig) {
...
+  if(dataApiLocation) {
```
Comment (sulabh-msft) [inline src/cli/commands/deploy/deploy.ts:38]: nit: `if(dataApiLocation) {` -> `if (dataApiLocation) {`

File: src/cli/commands/deploy/deploy.ts [hunk 3]
```diff
@@ -34,6 +34,11 @@ export async function deploy(options: SWACLIConfig) {
...
+    dataApiLocation = path.resolve(dataApiLocation);
```
Comment (sulabh-msft) [inline src/cli/commands/deploy/deploy.ts:39]: should we be logging in the console about the dataApiLocation

File: src/cli/commands/deploy/deploy.ts [hunk 4]
```diff
@@ -161,6 +166,7 @@ export async function deploy(options: SWACLIConfig) {
     appLocation,
     outputLocation: resolvedOutputLocation,
     apiLocation: resolvedApiLocation,
+    dataApiLocation
```
Comment (sulabh-msft) [inline src/cli/commands/deploy/deploy.ts:169]: Any change do we need to do in the readWorkflowFile for enabling it to handle dataApiLocation ?

File: src/cli/commands/deploy/deploy.ts (hunk 1 above)
Comment (thomasgauvin) [reply src/cli/commands/deploy/deploy.ts:37]: By convention, if the folder `swa-db-connections` is in the root directory, it gets deployed (no need to pass default args to the StaticSitesClient)

File: src/cli/commands/deploy/deploy.ts (hunk 2 above)
Comment (thomasgauvin) [inline src/cli/commands/deploy/deploy.ts:38]:
```suggestion
  if (dataApiLocation) {
```

Comment (thomasgauvin) [reply src/cli/commands/deploy/deploy.ts]: done! thanks

File: src/cli/commands/deploy/deploy.ts (hunk 3 above)
Comment (thomasgauvin) [reply src/cli/commands/deploy/deploy.ts:39]: Good catch thanks Sulabh, this & error logs/validation of path is missing, adding it

File: src/cli/commands/deploy/deploy.ts (hunk 4 above)
Comment (thomasgauvin) [reply src/cli/commands/deploy/deploy.ts:177]: Thanks Sulabh, I've adjusted this as well. I noticed that this feature currently broken while dev'ing and testing. I will let the bugfix take place in another PR.

**Comment by thomasgauvin** [comment]
From my testing, SWA CLI currently doesn't work well with extracting workflow config from gh action workflow files, needs some bug fixing there since the values are extracted but not properly used. That is to be addressed in a separate PR

File: src/cli/commands/deploy/deploy.ts (hunk 4 above)
Comment (sulabh-msft) [reply src/cli/commands/deploy/deploy.ts:177]: yup make sense. @sgollapudi77 please create a tracking task if we don't have it already.

File: src/core/utils/workflow-config.ts [hunk 5]
```diff
@@ -118,19 +119,24 @@ export function readWorkflowFile({ userWorkflowConfig }: { userWorkflowConfig?:
...
   // - output_location
+  // - data_api_location
 
   app_location = path.normalize(path.join(process.cwd(), app_location));
   if (typeof api_location !== "undefined") {
     api_location = path.normalize(path.join(process.cwd(), api_location || path.sep));
   }
   output_location = path.normalize(output_location);
   output_location = path.join(app_location, output_location);
+  if(typeof data_api_location !== "undefined") {
```
Comment (sulabh-msft) [inline src/core/utils/workflow-config.ts:137]:
```suggestion
  if (typeof data_api_location !== "undefined") {
```

**Review by sulabh-msft — APPROVED** [review]
Looks good. Left one nit comment.

File: src/cli/commands/deploy/deploy.ts (hunk 4 above)
Comment (sgollapudi77) [reply src/cli/commands/deploy/deploy.ts:177]: Added a task now Sulabh.

**Pull request merged**